docs/java-api-quickstart.md

```diff
@@ -38,37 +38,42 @@ The Hive catalog connects to a Hive metastore to keep track of Iceberg tables.
 You can initialize a Hive catalog with a name and some properties.
 (see: [Catalog properties](../configuration/#catalog-properties))
 
-**Note:** Currently, `setConf` is always required for hive catalogs, but this will change in the future.
+### Using a Hive catalog
+
+The Hive catalog connects to a Hive metastore to keep track of Iceberg tables.
+You can initialize a Hive catalog with a name and some properties.
+(see: [Catalog properties](../configuration/#catalog-properties))
 
 ```java
+import java.util.HashMap
+import java.util.Map
+
 import org.apache.iceberg.hive.HiveCatalog;
 
 HiveCatalog catalog = new HiveCatalog();
-catalog.setConf(spark.sparkContext().hadoopConfiguration());  // Configure using Spark's Hadoop configuration
 
 Map <String, String> properties = new HashMap<String, String>();
-properties.put("warehouse", "...");
-properties.put("uri", "...");
+properties.put("warehouse", "..."); // Override hive.metastore.warehouse.dir
+properties.put("uri", "..."); // Override hive.metastore.uris
```

Comment:
I don't agree with adding `Override` since the Hadoop configuration is now optional. These are the recommended and primary way to configure the warehouse and URI for Hive.